### ISSUE TITLE
[new rule] prefer blank line between tests or blocks

### ISSUE THREAD

**Issue by HaveSpacesuit** [jest-community/eslint-plugin-jest#1251]
Add a style rule to suggest adding a blank line between tests or blocks.

```ts
// INCORRECT
beforeAll(() => {
  /* ... */
}) // missing space after
describe('function a', () => {
  beforeEach(() => {
    /* ... */
  }) // missing space after
  it('returns one', () => {
    /* ... */
  }) // missing space after
  it('returns two', () => {
    /* ... */
  })
}) // missing space after
describe('function b', () => {
  it('returns three', () => {
    /* ... */
  }) // missing space after
  it('returns four', () => {
    /* ... */
  })
})

// CORRECT
beforeAll(() => {
  /* ... */
}) // space after

describe('function c', () => {
  beforeEach(() => {
    /* ... */
  }) // space after

  it('returns five', () => {
    /* ... */
  }) // space after

  it('returns six', () => {
    /* ... */
  })
}) // space after

describe('function d', () => {
  it('returns seven', () => {
    /* ... */
  }) // space after

  it('returns eight', () => {
    /* ... */
  })
})
```

I suppose the opposite could be enforced - for people who don't want spaces between blocks. Optionally, could add rules for only tests, only describe/before/after blocks, etc.

**Comment by G-Rath** [comment]
Is this not covered by [`eslint-plugin-jest-formatting`](https://github.com/dangreenisrael/eslint-plugin-jest-formatting)?

**Comment by HaveSpacesuit** [comment]
Thanks for the pointer! I didn't know about that package. Every time a new developer joins my team, I go searching for _more_ lint rules! 🤣🤣🤣
